src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoException.scala

```diff
@@ -0,0 +1,12 @@
+package org.broadinstitute.dsde.workbench.leonardo.model
+
+import akka.http.scaladsl.model.StatusCode
+import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchException, WorkbenchExceptionWithErrorReport}
+import org.broadinstitute.dsde.workbench.leonardo.errorReportSource
+import org.broadinstitute.dsde.workbench.model.ErrorReport._
+
+abstract class LeoException(message: String = null) extends WorkbenchException(message) {
```

Comment:
This is meant to be the base class of all Leo exceptions, right? There might be some cases where we _do_ want to return a cause/stack trace. Perhaps `LeoException` should take a `cause` constructer arg, and allow its subclasses to decide whether or not they want it.

Reply:
in general returning stacktraces is a security oopsie. if we're returning an exception derived from `LeoException` it's in order to be informative to the user. "real" (i.e. scary-unhandled) exceptions will end up at the more general exception handler in `complete()` and result in an autogenerated 500. we generally try to stamp those ones out asap.

Reply:
Fair, but we should make sure we still log the stack traces for debugging if we're suppressing them in the response. Idea: maybe include a UUID in the http response which could cross-reference to logs?

Reply:
Actually yeah, I agree with you. There are places in Rawls that we throw `RawlsException` because the db is in a messed up state and a stacktrace _is_ helpful to developers there. Thanks.